crosscorrelation.Rmd

```diff
@@ -0,0 +1,384 @@
+  for (i in 1:length(x)[[1]])
+  indexnum <- c(x[[i]]) 
+  indexnum1<-as.numeric(indexnum)
+return(indexnum1)
+}
+
+##main
+#
+system.time(values<-stringcnvert(df1$mgvalue,df2$mgvalue,3))#http://www.cs.ucr.edu/~eamonn/SAX.pdf alphabet size: 5-8 works better.
+#user  system elapsed 
+#   0.11    0.00    0.11 
+
+system.time(strngindex<-duplicateindex(values))
+validate<-extractindex(df1,df2,strngindex)
+seq1<-extractvalue(validate[3])
+seq2<-extractvalue(validate[4])
+
+df1_mgvalue<-patient[seq1,"mgvalue"]
+df2_mgvalue<-patient[seq2,"mgvalue"]
+compare<-data.frame(df1_mgvalue,df2_mgvalue)
+
+# par(mfrow=c(1,2))
+# plot.ts(compare$df1_mgvalue)
+# plot.ts(compare$df2_mgvalue)
+
+plot.zoo(cbind(compare$df1_mgvalue,compare$df2_mgvalue), 
+         plot.type = "multiple", 
+         col = c("red", "blue"))
+
+
+sum(compare$df1_mgvalue-compare$df2_mgvalue)
+
+
+```
+
+
+
+```{r}
+values<-stringcnvert(df3$mgvalue,df4$mgvalue,3 ) #http://www.cs.ucr.edu/~eamonn/SAX.pdf alphabet size: 5-8 works better. ##3 worked better for this run
+strngindex<-duplicateindex(values)
+validate<-extractindex(df3,df4,strngindex)
+seq1<-extractvalue(validate[3])
+seq2<-extractvalue(validate[4])
+
+df1_mgvalue<-patient[seq1,"mgvalue"]
+df2_mgvalue<-patient[seq2,"mgvalue"]
+compare<-data.frame(df1_mgvalue,df2_mgvalue)
+
+# par(mfrow=c(1,2))
+# plot.ts(compare$df1_mgvalue)
+# plot.ts(compare$df2_mgvalue)
+
+plot.zoo(cbind(compare$df1_mgvalue,compare$df2_mgvalue), 
+         plot.type = "multiple", 
+         col = c("red", "blue"))
+
+
+sum(compare$df1_mgvalue-compare$df2_mgvalue)
+```
+
+
+
+
+```{r}
+patient<-read.csv("duplicated0fe539475b52ae23f939d7dd2596cf8eb1e877edcea0478f2df73bb98bd5937c.csv")[,field]
+patient$mgvalue<-patient$value*18.01559
+length(unique(patient$uploadId)) #2
+length(unique(patient$id))==nrow(patient) ##True
+length(unique(patient$utcTime))# 8899
+anyDuplicated(patient$utcTime) #4366
+length(unique(patient$utcTime))==nrow(patient) #False
+
+
+
+df1<-subset(patient,patient$uploadId=="2f61322480c841fd8679fe81e94930b2")
+
+
+df2<-subset(patient,patient$uploadId=="c05970591b404518a1cbd64595d628e5")
+  
+
+
+```
+
+
+```{r}
+values<-stringcnvert(df1$mgvalue,df2$mgvalue,8)
+strngindex<-duplicateindex(values)
+validate<-extractindex(df1,df2,strngindex)
+seq1<-extractvalue(validate[3])
+seq2<-extractvalue(validate[4])
+
+df1_mgvalue<-patient[seq1,"mgvalue"]
+df2_mgvalue<-patient[seq2,"mgvalue"]
+compare<-data.frame(df1_mgvalue,df2_mgvalue)
+
+
+
+plot.zoo(cbind(compare$df1_mgvalue,compare$df2_mgvalue), 
+         plot.type = "multiple", 
+         col = c("red", "blue"))
+
+sum(compare$df1_mgvalue-compare$df2_mgvalue)
+
+
+```
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+#cross correlation
+```{r}
+corre<-function(x,y,lagsize){
+  corr<-ccf(x$mgvalue,y$mgvalue,max.lag=lagsize,plot=TRUE)
+  max.value<-max(corr$acf)
+  max.lag<-corr$lag[which(corr$acf==max.value)]
+  xlength<-length(x$mgvalue)
+  ylength<-length(y$mgvalue)
+  if ((max.lag==0) & (xlength<ylength)) {
```

Comment:
Please add a small comment to each of these conditions describing the underlying intention/what we're checking for with respect to cross-correlation